packages/@aws-cdk/pipelines/lib/actions/publish-assets-action.ts

```diff
@@ -124,13 +129,15 @@ export class PublishAssetsAction extends CoreConstruct implements codepipeline.I
       role: props.role,
     });
 
-    const rolePattern = props.assetType === AssetType.DOCKER_IMAGE
-      ? 'arn:*:iam::*:role/*-image-publishing-role-*'
-      : 'arn:*:iam::*:role/*-file-publishing-role-*';
-
     project.addToRolePolicy(new iam.PolicyStatement({
       actions: ['sts:AssumeRole'],
-      resources: [rolePattern],
+      resources: [Fn.sub(props.roleArnsForCdkAssets)],
+    }));
+    // allow assuming any role in a different account
+    // because custom names may have been used
+    project.addToRolePolicy(new iam.PolicyStatement({
+      actions: ['sts:AssumeRole'],
+      notResources: [Fn.sub('arn:${AWS::Partition}:iam::${AWS::AccountId}:role/*')],
```

Comment:
This line confuses me. Wildcards should not be necessary if we know the entire list?

Reply:
We know the entire list for the current account, which is what's allowed in the statement above this one.

We also need to allow assuming the role in another account though, and we don't know what it might be called in another account, because they might be named differently there.

This `NotResource` is meant to result in allowing assuming all roles, excluding any role in the current account.

Reply:
Ah so `props.publishRoleArn` could already be an arn to another account? So we could just remove this?